src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalyzer.cs

```diff
@@ -23,12 +23,12 @@ public abstract class DiagnosticAnalyzer
 
         public sealed override bool Equals(object obj)
         {
-            return ReferenceEqualityComparer.Instance.Equals(this, obj);
+            return ReferenceEquals(this, obj);
```

Comment:
It was extremely roundabout reference compare before. I was not brave to simplify it that much :-)
Will change.